### PR TITLE
default split for train_pipeline and scoring

### PR DESCRIPTION
**Pull request by BenoitHardier**

### CODE REVIEW

File: openstef/model_selection/model_selection.py [hunk 1]
```diff
@@ -173,9 +174,8 @@ def split_data_train_validation_test(
         test_data = data_[start_date_test:]
         train_val_data = data_[:start_date_test]
     else:
-        start_date_val = start_date + np.round(number_indices * test_fraction) * delta
-        test_data = data_[:start_date_val]
-        train_val_data = data_[start_date_val:]
+        test_data = data_.copy(deep=True)
```
Comment (JanMaartenvanDoorn) [inline openstef/model_selection/model_selection.py:177]: I remember disusing this but on a closer look I do not understand this anymore. For the split_test_train_validation data I would expect three different datasets to come out (may also be empty datasets).

What is the problem again with just using `pd.concat([train_data, validation_data])` on the place were you want to use the test_data_set?

File: openstef/model_selection/model_selection.py [hunk 2]
```diff
@@ -141,6 +141,7 @@ def split_data_train_validation_test(
         - Test data.
 
     """
+    test_fraction = test_fraction if back_test else 0
```
Comment (JanMaartenvanDoorn) [inline openstef/model_selection/model_selection.py:144]: This seems logical

File: openstef/pipeline/train_model.py [hunk 3]
```diff
@@ -199,14 +199,13 @@ def train_model_pipeline_core(
 
     # Check if new model is better than old model
     if old_model:
-        combined = pd.concat([train_data, validation_data]).reset_index(drop=True)
         # skip the forecast column added at the end of dataframes
         if pj.save_train_forecasts:
-            combined = combined.iloc[:, :-1]
+            test_data = test_data.iloc[:, :-1]
```
Comment (JanMaartenvanDoorn) [inline openstef/pipeline/train_model.py:204]: See question above

File: test/unit/pipeline/test_train.py [hunk 4]
```diff
@@ -73,10 +73,9 @@ def test_split_data_train_validation_test_stratification_no_backtest(self):
             delta=2,
         )
```
Comment (JanMaartenvanDoorn) [inline test/unit/pipeline/test_train.py:74]: Nice improvements in this file, make is more clear!

**Review by JanMaartenvanDoorn — COMMENTED** [review]
Upon closer look I think I still do not understand why we should change the test_data set. Can you explain a bit more about this?

File: openstef/model_selection/model_selection.py (hunk 1 above)
Comment (BenoitHardier) [reply openstef/model_selection/model_selection.py:177]: The main reason is related to the use of custom split function. If you implement your own split procedure it seems very akward to design it with this concat operation in mind. Moreover it seems very sensible to compute scores/metrics on the test set.   
My changes do not affect the fact that the split function returns 3 datasets (they are just overlapping when `back_test=False`).   
The second issue is the fact that this concat can introduce weird behaviours when the order of train and val are shuffled. This has no consequences for pointwise models like xbgm/lightgbm because lag features has been computed before splitting. But with ARIMA (or other ar models like prophete) it can be very painfull, that why we prefer to keep this operation behind the split function.
